redhat-cop/infra-ansible
# Update confluence role to use new inventory and shared variables

### What does this PR do?
This PR updates the `manage-confluence` role to use an updated inventory structure. Previously `confluence` existed outside of the `atlassian` structure. This now wraps all atlassian products underneath this new inventory structure. It also allows the `manage-confluence` role to share variables between the upper level `atlassian` dictionary and the lower level `atlassian.confluence.source` and `atlassian.confluence.destination` dictionaries.

### How should this be tested?
This can be tested by running the tests underneath `roles/manage-confluence-space/tests`. This provides the following tests:
- `vars_all`: `username, password, url` provided in `atlassian, atlassian.confluence.source`, and `atlassian.confluence.destination`
- `vars_atlassian`: `username, password,url` provided by `atlassian`
- `vars_no_atlassian`: `username,password,url` provided by `atlassian.confluence.source` and `atlassian.confluence.destination`
- `vars_no_source`: `username,password,url` provided by `atlassian` and `atlassian.confluence.destination`
- `vars_no_destination`: `username,password,url` provided by `atlassian` and `atlassian.confluence.source`

You can run these tests by running the following:

`ansible-playbook -i inventory playbook.yml` and providing the appropriate vars file in `playbook.yml`. You'll also need to make sure that appropriate values (i.e. real url, credentials, etc.) are provided in your chosen vars file (ex. vars_atlassian).

### Is there a relevant Issue open for this?
#246 and completes resolution of #242 

### People to notify
cc: @redhat-cop/infra-ansible
